Userland/Applications/HexEditor/HexEditor.cpp

```diff
@@ -500,7 +500,7 @@ ErrorOr<void> HexEditor::hex_mode_keydown_event(GUI::KeyEvent& event)
         if (m_document->size() == 0)
             return {};
 
-        VERIFY(m_position <= m_document->size());
+        VERIFY(m_position < m_document->size());
```

Comment:
Hm, interesting. I get those crashes every time. Basically I go to either hex/text mode, press page down and type anything.

In text mode it crashes on `VERIFY`, in `HexEditor::text_mode_keydown_event` method, it is because current cursor's position exceeds document's size. This is why I've done the equivalent change to VERIFY in `hex_mode_keydown_event` - in hex mode it crashes further, in `HexDocumentMemory::get` method, because VERIFY didn't catch it.

As you suggested - I also added VERIFY for size in `move_and_update_cursor_by`, in a separate commit.